src/main/kotlin/ai/whylabs/services/whylogs/core/WhyLogsController.kt

```diff
@@ -22,17 +25,22 @@ const val DatasetIdTag = "datasetId"
 const val OrgIdTag = "orgId"
 
 class WhyLogsController(
-    private val envVars: IEnvVars = EnvVars(),
+    private val envVars: IEnvVars = EnvVars.instance,
     private val profileManager: WhyLogsProfileManager = WhyLogsProfileManager(envVars = envVars),
 ) {
     private val logger = LoggerFactory.getLogger(javaClass)
 
     fun preprocess(ctx: Context) {
         // This shouldn't actually happen. Swagger will take care of the validation.
         val apiKey = ctx.header(apiKeyHeader)
-            ?: throw IllegalArgumentException("Missing api key in request")
+
+        if (apiKey == null) {
```

Comment:
Good question. I'm actually just going to delete this. I added it because of some issue that I can't repro anymore and the other if statement is clearly sufficient.